### PR TITLE
feat: support edge runtime

### PR DESCRIPTION
**Pull request by promer94**
add environment support for https://github.com/vercel/edge-runtime

## limitations
`eval` and `new Function()` still works in vitest edge environment. 
#1470 might help to solve this problem
```js
// ❌ Fail
// Error name:    "EvalError"
// Error message: "Code generation from strings disallowed for this context"
it('should return the correct value', () => {
  let val = eval('2 + 2');
  expect(val).toBe(4);
});

```

### CODE REVIEW

File: docs/config/index.md [hunk 1]
```diff
@@ -153,13 +153,14 @@ export default defineConfig({
 
 ### environment
 
-- **Type:** `'node' | 'jsdom' | 'happy-dom'`
+- **Type:** `'node' | 'jsdom' | 'happy-dom' | 'edge'`
```
Comment (antfu) [inline docs/config/index.md:156]:
```suggestion
- **Type:** `'node' | 'jsdom' | 'happy-dom' | 'edge-runtime'`
```

`edge` feels too general without knowing the context, maybe better to be `edge-runtime`?

**Review by antfu — APPROVED** [review]
I like this in general. Just a few details need to discuss

File: packages/vitest/src/integrations/env/index.ts [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
 export const environments = {
   node,
   jsdom,
   'happy-dom': happy,
+  'edge-runtime': edge,
+}
+
+export const envs = Object.keys(environments)
+
+export const envNpm = {
```
Comment (antfu) [inline packages/vitest/src/integrations/env/index.ts:15]:
```suggestion
export const envPackageNames: Record<keyof typeof environments, string> = {
```

File: packages/vitest/src/node/cli-api.ts [hunk 3]
```diff
@@ -37,7 +38,8 @@ export async function startVitest(cliFilters: string[], options: CliOptions, vit
   }
 
   if (ctx.config.environment && ctx.config.environment !== 'node') {
-    if (!await ensurePackageInstalled(ctx.config.environment)) {
+    const packName = envNpm[ctx.config.environment]
```
Comment (antfu) [inline packages/vitest/src/node/cli-api.ts:41]:
```suggestion
    const packageName = envNpm[ctx.config.environment]
```

**Review by antfu — APPROVED** [review]

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Should we also add `@edge-runtime/vm` to peerDependencies? We have all other envs there

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**